src/control/control.cpp

```diff
@@ -38,6 +38,7 @@ ControlDoublePrivate::ControlDoublePrivate()
                   Stat::SAMPLE_VARIANCE | Stat::MIN | Stat::MAX),
           // default CO is read only
           m_confirmRequired(true) {
+    m_value.setValue(0.0);
```

Comment:
This silences just the valgrind warning and make the bug itself invisible for code analysis. The root cause must be somewhere in the use of the overloaded classes.

Reply:
This is the value of the constant default ControlObject that is used whenever there is no real control object. 
This has been introduced to get around many null checks when dealing with control objects. 
The original issue happens here: https://github.com/mixxxdj/mixxx/blob/7d69c32f7cbc82aa1f8bbfb5a83f406e3f08e682/src/widget/wwidgetstack.cpp#L157

Reply:
Than you must set the correct value (might be 0 here), in the class, where the original issue occured.
Otherwise you make Valgrind blind for bugs of the same kind in other classes. And it depends on the context, if 0 is the correct value there.

Reply:
We only have a single Default Control object for all, and it is const: 
https://github.com/mixxxdj/mixxx/blob/7d69c32f7cbc82aa1f8bbfb5a83f406e3f08e682/src/control/control.cpp#L192
So it should have a reasonable default of 0 instead of a random value.

Reply:
The only important point is: **Never initialize with dummy values!**
Which implies to not initialize in the base class, do this in /src/widget/wwidgetstack.cpp instead.
Otherwise Valgrind will be blind for "Use Of Uninitialiezd Variable" bugs introduced in future.
Valgrind can not distinguish between initialization with dummy values and intialize with real values. But it can detect code pathes, were the variable is used before initialization. Therefore always initializes with the real value, which is correct in the specific context - than Valgrind, ASAN and statical code analysis tools like cppcheck will detect the real bugs in other paths.

Reply:
I can confirm the rule in general. But in this case the whole ContrObject is a singleton read only dummy. There is no point of a business logic value initialization and you demand is not possible here. 
The control object system has already warnings if a control uses the default CO unintentionally.

Reply:
If there are already warnings in place, this is of cause Ok here.